packages/rest/src/controllers/proofs/ProofController.ts

```diff
@@ -0,0 +1,240 @@
+
+@JsonController("/proofs")
+@Service()
+export class ProofController {
+  @Inject()
+  private agent: Agent;
+  private proofUtils: ProofUtils;
+
+  public constructor(agent: Agent) {
+    this.agent = agent;
+    this.proofUtils = new ProofUtils(agent);
+  }
+
+  /**
+   * Retrieve proof record by proofRecordId
+   */
+  @Get("/:proofRecordId")
+  public async getProofById(@Param("proofRecordId") proofRecordId: string) {
+    const proofRecord = await this.agent.proofs.getById(proofRecordId);
+
+    if (!proofRecord) {
+      throw new NotFoundError(`proof record with proofRecordId "${proofRecordId}" not found.`);
+    }
+
+    return proofRecord;
+  }
+
+  /**
+   * Retrieve proof record by threadId
+   */
+  @Get("/thread/:threadId")
+  public async getProofByThreadId(@Param("threadId") threadId: string) {
+    const proofRecord = await this.proofUtils.getProofByThreadId(threadId);
+
+    if (!proofRecord) {
+      throw new NotFoundError(`proof record for threadId "${threadId}" not found.`);
+    }
+
+    return proofRecord;
+  }
+
+  /**
+   * Retrieve all ProofRecords
+   */
+  @Get("/")
+  public async getAllProofs() {
+    const proofs = await this.agent.proofs.getAll();
+
+    return proofs;
+  }
+
+  /**
+   * Initiate a new presentation exchange as prover by sending a presentation proposal message
+   * to the connection with the specified connection id.
+   */
+  @Post("/:connectionId/propose-proof")
+  public async proposeProof(@Param("connectionId") connectionId: string, @Body() proposal: ProofProposelRequest) {
+    try {
+      const { attributes, predicates, ...restProposal } = proposal;
+
+      const newPresentationProposal = {
+        ...restProposal,
+        attributes: attributes,
+        predicates: predicates,
+      };
+      const presentationProposal = JsonTransformer.fromJSON(newPresentationProposal, PresentationPreview);
+
+      const proofRecord = await this.agent.proofs.proposeProof(connectionId, presentationProposal, {
+        comment: proposal.comment,
+      });
+
+      return proofRecord;
+    } catch (error) {
+      if (error instanceof RecordNotFoundError) {
+        throw new NotFoundError(`connection with connectionId "${connectionId}" not found.`);
+      }
+      throw new InternalServerError(`something went completely wrong: ${error}`);
+    }
+  }
+
+  /**
+   * Accept a presentation proposal as verifier (by sending a presentation request message) to the connection
+   * associated with the proof record.
+   */
+  @Post("/:proofRecordId/accept-proposal")
+  public async acceptProposal(@Param("proofId") proofRecordId: string, @Body() proposal: AcceptProofProposalRequest) {
+    const proofRecord = await this.agent.proofs.acceptProposal(proofRecordId, proposal);
+
+    return proofRecord;
+  }
+
+  /**
+   * Creates a presentation request not bound to any proposal or existing connection
+   */
+  @Post("/request-proof")
+  public async requestProofOutOfBand(
```

Comment:
Maybe something like: `/oob-request` (not sure, I'm just wondering whether request-proof can be confusing)